src/com/gfredericks/schpec/number.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.number
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(defn real?
+  [x]
+  (and (number? x)
+       (case x
+         Double/POSITIVE_INFINITY false
+         Double/NEGATIVE_INFINITY false
+         Double/NaN false
+         true)))
+
+(s/def ::real
+  (s/spec real?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
```

Comment:
No, it's intended to apply to all real number values, which as far as I can reason, include all things clojure considers numbers except these three corner cases. I confess, I don't see any mismatch between this predicate and the mathematical domain of real numbers. Is it because there are irrational numbers (e.g. pi) that don't have precise representations?

Clojure core already provides `double?`. My intent here is to provide predicates and specs for numbers in terms of their domain (e.g. real numbers, numbers that can be precisely represented as decimals with such and such precision), not their actual types.